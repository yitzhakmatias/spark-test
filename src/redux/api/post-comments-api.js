import defaultAxios from 'axios';

const axios = defaultAxios.create({
  baseURL: 'https://jsonplaceholder.typicode.com/',
  headers: {'Content-Type': 'application/json'},
});

export const getAllPosts = async () => {
  try {
    const posts = await axios.get('posts');

    return posts.data;
  } catch (err) {
    return console.error(err);
  }
};
export const getAllComments = async (id) => {
  try {
    const comments = await axios.get( `comments?postId=${id}`);

    return comments.data;
  } catch (err) {
    return console.error(err);
  }
};
// Create New Comment
export const createNewComment = async (comment) => {
  try {
    return comment;
  } catch (err) {
    return console.error(err);
  }
};

