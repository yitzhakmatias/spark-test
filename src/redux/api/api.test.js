// @flow
import * as actions from './post-comments-api.js';

describe('user actions', () => {
    test('getAllComments', () => {
        expect(actions.getAllComments()).toMatchSnapshot();
    });

    test('getAllPosts', () => {
        expect(actions.getAllPosts()).toMatchSnapshot();
    });


    test('createNewComment', () => {
        const payload = {
            postId: 2,
            name: 'Czechmademan',
            body: 'testsetsetset'
        }
        expect(actions.createNewComment(payload)).toMatchSnapshot();
    });

});
