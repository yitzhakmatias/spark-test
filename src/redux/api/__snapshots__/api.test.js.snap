// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user actions createNewComment 1`] = `Promise {}`;

exports[`user actions getAllComments 1`] = `Promise {}`;

exports[`user actions getAllPosts 1`] = `Promise {}`;
